### PR TITLE
Clarify the runtime directory location in installation docs

### PR DESCRIPTION
**Pull request by kazimir-malevich**
https://github.com/helix-editor/helix/issues/6607

### CODE REVIEW

File: book/src/install.md
Comment (David-Else) [inline book/src/install.md:176]: I don't think `<HELIX_REPO_ROOT>` is as clear as `/home/user-name/src/helix`, it looks like an environment variable, but it does not exist. I agree that just saying `src` might cause confusion, but please note the Windows section also refers to `src`:

```
mklink /D runtime "%userprofile%\src\helix\runtime"
```
There is an assumption that the user knows this refers to the location they cloned Helix, but that could be improved, I think the best solution is to add something like:

> ## [Building from source](https://docs.helix-editor.com/install.html#building-from-source)

> ### Choosing a location for the Helix source code

> If you don't already have a directory on your computer for source code, please create one and `CD` into it, this is where Helix's source code will live. The examples in this documentation assume installation into either `~/src/` on Linux/Mac,  or `%userprofile%\src\helix\runtime` on Windows.

I don't have time to get the wording perfect, but this is roughly what I would do.

Comment (kazimir-malevich) [reply book/src/install.md:176]: Hi @David-Else I am utterly convinced of your sentiments, however I've looked into the Linux filesystem hierarchy and maybe we should be suggesting users install the helix source code in **/opt** or **usr/local/src**. And what about pasting the bash script you gave me into the docs and I make some changes? Some Unix users prefer compiling from source and I was hitting all sorts of hurdles with Debian and pre packaged binaries (old compilers etc)? You're probably taking a well-deserved break so I will go ahead.

Comment (kazimir-malevich) [reply book/src/install.md]: Hi @David-Else I don't like the idea of following the UNIX FSH and using **sudo** to install **Helix** when it can be installed as a regular user. I have thus changed the documentation to create  a **$HOME/src/** dir.

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:167]: I don't think this is necessary to add. Compiling the tree-sitter grammars with clang works just as well and this makes gcc-c++ look required.

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:140]: Since we're not going to use FHS guideless I don't think we need to mention it

**Review by the-mikedavis — COMMENTED** [review]
The `HELIX_RUNTIME` and symlink options both allow you to install the Helix repo anywhere. Giving instructions that use `~/src` explicitly makes it seem like it must be installed there which isn't the case and I think different people like to put their source code in different places.

So I think the runtime instructions for linux should just be more explicit about "the directory where the Helix repository is cloned." Mentioning `~/src` as an example is ok but I think we should avoid hard-coding it in the instructions.

File: book/src/install.md
Comment (kazimir-malevich) [reply book/src/install.md]: Suggestions adopted and committed @the-mikedavis

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:176]:
```suggestion
# Replace "/home/user-name/src/helix" with the path to where you cloned
# the Helix repository.
HELIX_RUNTIME=/home/user-name/src/helix/runtime
```

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:182]: I believe the link here will not resolve correctly unless repository_directory is an absolute path. The PWD version should work as long as you have `cd`d into the clone directory.

```suggestion
# Make sure you are in the Helix repository clone's directory.
ln -s $PWD/helix/runtime ~/.config/helix/runtime
```

**Review by the-mikedavis — COMMENTED** [review]
The `repository_directory` placeholder looks like a regular directory name to me and I think that makes this more confusing. Instead I think we should keep the instructions mostly as-is and focus on the confusing parts of the `HELIX_RUNTIME` and `ln` steps: replacing the example absolute path and ensuring that we have `cd`d into the clone's directory.

**Comment by David-Else** [comment]
Please see https://github.com/helix-editor/helix/pull/6809 , I have not addressed any of the things mentioned here on purpose, but please rebase when/if my PR gets merged as it changes some things around.

**Comment by kazimir-malevich** [comment]
As of time speaking #6809 is still Open. I have attempted to synthesise directions from both @David-Else and @the-mikedavis.

File: book/src/install.md
Comment (David-Else) [inline book/src/install.md:154]: This bit has now been addressed in https://github.com/helix-editor/helix/pull/6809

File: book/src/install.md
Comment (David-Else) [inline book/src/install.md:139]: I like it! In https://github.com/helix-editor/helix/pull/6809 there is now a list format:

1. Clone the repository:

You could put your new paragraph under that, or turn `Clone the repository:` and `Compile from source:` into sub headings... I think I had them subheadings originally, but the maintainers didn't like it, @the-mikedavis will know best :)

Comment (kazimir-malevich) [reply book/src/install.md:139]: @David-Else Should we merge these to PRs or should I wait until #6809 is merged into master?

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:184]: I think we should continue to use `~/src` for these from the new intro in the building from source section. `/home/<name>` might be confusing for macos users since home directories are under `/Users/<name>` in macos

**Review by the-mikedavis — COMMENTED** [review]
#6809 has been merged. Could you rebase or merge master to put this on top of those changes?

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:185]: The changes here are just spacing and should be reset

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:202]:
```suggestion
HELIX_RUNTIME=~/src/helix/runtime
```

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:208]:
```suggestion
ln -s ~/src/helix/runtime ~/.config/helix/runtime
```

(just an extra space)

File: book/src/install.md
Comment (the-mikedavis) [inline book/src/install.md:187]:
```suggestion
This command will create the `hx` executable and construct the tree-sitter
grammars in the local `runtime` folder.
```

(we prefer to avoid cosmetic changes when possible, to keep the diff small)

**Comment by kazimir-malevich** [comment]
@David-Else Do I need to do anything more regarding this PR?

File: book/src/install.md
Comment (gabydd) [inline book/src/install.md:161]: To match with the Linux instructions I think we should drop the ``helix\runtime`` and have it just be `%userprofile%\src\`

**Review by gabydd — CHANGES_REQUESTED** [review]

File: book/src/install.md
Comment (kazimir-malevich) [reply book/src/install.md]: Good call @gabydd

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**
